storage/static.go

```diff
@@ -91,8 +92,9 @@ type staticPasswordsStorage struct {
 func WithStaticPasswords(s Storage, staticPasswords []Password) Storage {
 	passwordsByEmail := make(map[string]Password, len(staticPasswords))
 	for _, p := range staticPasswords {
-		p.Email = strings.ToLower(p.Email)
-		passwordsByEmail[p.Email] = p
+		//Enable case insensitive email comparison.
+		lowerEmail := strings.ToLower(p.Email)
+		passwordsByEmail[lowerEmail] = p
```

Comment:
@rithujohn191 right, but prior to this that config would not have errored, if we error now we might break someone during a dex upgrade.